test/Microsoft.ML.Functional.Tests/Validation.cs

```diff
@@ -51,6 +53,42 @@ public void CrossValidation()
                 Common.AssertMetrics(result.Metrics);
         }
 
+        [Fact]
+        public void AutoFitRankingCVTest()
+        {
```

Comment:
**CI error** I think this test is failing on the CI on Windows x86 because we don't have LightGBM on x86. Please try to add a [LightGBMFact] (I think that's called) to avoid this failure.

**Suggestion** Rename this test to only RankingCVTest, as no "AutoFit" is done (i.e. it doesn't run anything related to AutoML). Thanks.

Reply:
`LightGBMFact` is not enabled in functional tests. I'm not familiar with functional tests, so I'll defer adding a `LightGBMFact`. If you think it is as simple as adding the attribute set of classes to functional tests, I can do it here or on another PR.

Reply:
I think you'd only need to move the `LightGBMFact` into the `Microsoft.ML.TestFrameworkCommon` project, (e.g., the `OnnxFactAttribute` is in TestFrameWorkCommon, and that's how we can use it on FunctionalTests). Maybe @frankdong could tell use if there's something else to do.

An alternative would be to simply change the LightGBM trainer for another ranking trainer (this CV Ranking API is general purpose, and it should work with _any_ ranking trainer)

---
In reply to: [452954648](https://github.com/dotnet/machinelearning/pull/5263#discussion_r452954648) [](ancestors = 452954648)